src/mds/Locker.cc

```diff
@@ -3322,7 +3322,8 @@ void Locker::_update_cap_fields(CInode *in, int dirty, const cref_t<MClientCaps>
       else
 	pi->inline_data.free_data();
     }
-    if ((dirty & CEPH_CAP_FILE_EXCL) && atime != pi->atime) {
+    if (((dirty & CEPH_CAP_FILE_EXCL) && atime != pi->atime) ||
+	((dirty & CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR) && atime > pi->atime)) {
```

Comment:
On second thought, I think we want to leave Fw out of this. The atime is all about reads, so we should probably only look at Fr caps.

Reply:
this does not work.   

When client has CEPH_CAP_FILE_EXCL or CEPH_CAP_FILE_WR, lock state of filelock allow force_wrlock.  But it may not  be true if client only CEPH_CAP_FILE_RD. 

If client does not have CEPH_CAP_FILE_EXCL or CEPH_CAP_FILE_WR, it needs to use setattr request to update atime.

Reply:
Ouch. The atime is all about reads, so I don't really get the rationale for tying it to Fw caps. Regardless, I'll go ahead and drop this PR since this won't work.